src/NativePresets/CMakeLists.txt

```diff
@@ -0,0 +1,70 @@
+if(NOT ENABLE_PRESETS OR NOT ENABLE_NATIVE_PRESETS)
+    return()
+endif()
+
+# Using a macro to reduce repetition.
+macro(add_native_preset name)
+    if(NOT CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        add_library(${name}_static STATIC ${name}.cpp)
+
+        set_target_properties(${name}_static PROPERTIES
+                OUTPUT_NAME ${name}
+                )
+
+        target_link_libraries(${name}_static
+                PRIVATE
+                projectM_static
+                GLM::GLM
+                )
+    
+        install(TARGETS ${name}_static
+                LIBRARY DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                RUNTIME DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                ARCHIVE DESTINATION "${PROJECTM_DATADIR_PATH}/presets"
+                COMPONENT Presets
+                )
+    endif()
+
+    add_library(${name}_shared SHARED ${name}.cpp)
```

Comment:
This is unconventional to have separate targets for static and shared libraries. Please use one target and let the user choose how to link with BUILD_SHARED_LIBS.

Reply:
Totally agree.

Reason for this is: that is just the first step in adding CMake scripts, so the initial goal was to recreate the same build output/file layout as the autoconf-based builds, which actually builds both libraries (even the native presets) and don't has an option to control that.

This makes it easier for now to find any issues left from porting the build. After that, it's just a few small changes to use these CMake-standard schemes, and will definitely be done as in most use cases, only the shared libraries will be required.

Reply:
Okay. That seems like a good plan to me. :)